pkg/extension/sumologicextension/extension.go

```diff
@@ -200,104 +169,161 @@ func (se *SumologicExtension) Start(ctx context.Context, host component.Host) er
 	return nil
 }
 
-// handleCredentials retrieves the credentials for the collector.
-// It does so by checking the local credentials store and by validating those credentials.
-// In case they are invalid or are not available through local credentials store
-// then it tries to register the collector using the provided access keys.
-func (se *SumologicExtension) handleCredentials(ctx context.Context, host component.Host) (credentials.CollectorCredentials, bool, error) {
-	colCreds, registrationDone, err := se.getCredentials(ctx)
-	if err != nil {
-		return credentials.CollectorCredentials{},
-			registrationDone,
-			err
+// Shutdown is invoked during service shutdown.
+func (se *SumologicExtension) Shutdown(ctx context.Context) error {
+	se.closeOnce.Do(func() { close(se.closeChan) })
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	default:
+		return nil
+	}
+}
+
+func (se *SumologicExtension) validateCredentials(
+	ctx context.Context,
+	host component.Host,
+	colCreds credentials.CollectorCredentials,
+) error {
+	se.logger.Info("Validating collector credentials...",
+		zap.String(collectorIdField, se.registrationInfo.CollectorCredentialId),
+		zap.String(collectorNameField, se.registrationInfo.CollectorName),
+	)
+
+	if err := se.injectCredentials(colCreds, host); err != nil {
+		return err
```

Comment:
Do we actually need this here? It doesn't look like it does anything other than setting the client, which we're explicitly using for the heartbeat anyway.

Reply:
It is needed as it sets the credentials in `registrationInfo` and the client (as you mentioned).